src/components/radio/radio.ts

```diff
@@ -69,7 +69,10 @@ export class MdRadioGroup implements AfterContentInit, ControlValueAccessor {
   private _value: any = null;
 
   /** The HTML name attribute applied to radio buttons in this group. */
-  private _name: string = null;
+  private _name: string = `md-radio-group-${_uniqueIdCounter++}`;
+
+  /** @internal */
+  nameEmitter: EventEmitter<string> = new EventEmitter<string>();
```

Comment:
Instead of this, could each radio pull the group's name in its own `ngOnInit`, and then the group can update them all whenever its name changes (in the setter)?